internal/tiltfile/tiltfile_state.go

```diff
@@ -1476,9 +1488,37 @@ func (s *tiltfileState) translateLocal() ([]model.Manifest, error) {
 		result = append(result, m)
 	}
 
+	for _, manifest := range result {
+		s.warnDuplicateYamlEntities(manifest)
+	}
+
 	return result, nil
 }
 
+//this deals with alerting the user if there are multiple instances of a
+//the same yaml entity (same name:kind:namespace:group) which may have been
+//defined in the same way or in multiple ways.
+func (s *tiltfileState) warnDuplicateYamlEntities(m model.Manifest) {
+	deployTarget := m.K8sTarget()
+	displayNames := deployTarget.DisplayNames
+
+	duplicates := make(map[string]int)
+
+	//when the same entity appears more than once, Tilt handles it by storing it as name:kind:namespace:group:n where n is
+	//the nth repeat of that entity, so as to not cause execution to halt. In the future, it might be better to surface an
+	//error about the duplicated resource closer to the actual manifest assembly.
+	for _, name := range displayNames {
+		tempName := strings.Split(name, ":")
+		if len(tempName) > 4 { //this means there's an n which was appended at the manifest assembly because of multiple same entities present
+			duplicates[strings.Join(tempName[:4], ":")]++
+		}
+	}
+
+	for entity := range duplicates {
+		s.logger.Warnf("The following YAML Entity has been duplicated: " + entity)
```

Comment:
Not sure how helpful this error message will be to the user--what about something like:
```
"Resource %s defines k8s entity %s more than once. Only one definition will be applied to your cluster; to avoid unexpected behavior, please delete the redundant YAML."
```
or??? 🤷